pkg/kn/commands/common/human_readable_flags.go

```diff
@@ -8,74 +8,62 @@
-	"k8s.io/cli-runtime/pkg/genericclioptions"
-	"time"
 )
 
-// ServiceGetFlags composes common printer flag structs
-// used in the Get command.
-type ServiceGetFlags struct {
-	GenericPrintFlags  *genericclioptions.PrintFlags
-	HumanReadableFlags *HumanPrintFlags
+// HumanPrintFlags provides default flags necessary for printing.
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type HumanPrintFlags struct {
+	//TODO: Add more flags as required
 }
 
-// AllowedFormats is the list of formats in which data can be displayed
-func (f *ServiceGetFlags) AllowedFormats() []string {
-	formats := f.GenericPrintFlags.AllowedFormats()
-	formats = append(formats, f.HumanReadableFlags.AllowedFormats()...)
-	return formats
+// AllowedFormats returns more customized formating options
+func (f *HumanPrintFlags) AllowedFormats() []string {
+	// TODO: Add more formats eg: wide
+	return []string{""}
 }
 
-// ToPrinter attempts to find a composed set of ServiceGetFlags suitable for
-// returning a printer based on current flag values.
-func (f *ServiceGetFlags) ToPrinter() (hprinters.ResourcePrinter, error) {
-	// if there are flags specified for generic printing
-	if f.GenericPrintFlags.OutputFlagSpecified() {
-		p, err := f.GenericPrintFlags.ToPrinter()
-		if err != nil {
-			return nil, err
-		}
-		return p, nil
-	}
-	// if no flags specified, use the table printing
-	p, err := f.HumanReadableFlags.ToPrinter()
-	if err != nil {
-		return nil, err
-	}
+// ToPrinter receives returns a printer capable of
+// handling human-readable output.
+func (f *HumanPrintFlags) ToPrinter() (hprinters.ResourcePrinter, error) {
+	p := hprinters.NewTablePrinter(hprinters.PrintOptions{})
+	// Add the column definitions and respective printing functions for service get command
+	ServiceGetHandlers(p)
+	// Add the column definitions and respective printing functions for revision get command
+	RevisionGetHandlers(p)
 	return p, nil
 }
 
 // AddFlags receives a *cobra.Command reference and binds
-// flags related to humanreadable and template printing.
-func (f *ServiceGetFlags) AddFlags(cmd *cobra.Command) {
-	f.GenericPrintFlags.AddFlags(cmd)
-	f.HumanReadableFlags.AddFlags(cmd)
+// flags related to human-readable printing to it
+func (f *HumanPrintFlags) AddFlags(c *cobra.Command) {
+	//TODO: Add more flags as required
 }
 
-// NewGetPrintFlags returns flags associated with humanreadable,
-// template, and "name" printing, with default values set.
-func NewServiceGetFlags() *ServiceGetFlags {
-	return &ServiceGetFlags{
-		GenericPrintFlags:  genericclioptions.NewPrintFlags(""),
-		HumanReadableFlags: NewHumanPrintFlags(),
-	}
+// NewHumanPrintFlags returns flags associated with
+// human-readable printing, with default values set.
+func NewHumanPrintFlags() *HumanPrintFlags {
+	return &HumanPrintFlags{}
 }
 
 // ServiceGetHandlers adds print handlers for service get command
```

Comment:
This should live with the service package, in the refactor.

Reply:
OK. Will try and address this with next update